src/controllers/login.js

```diff
@@ -0,0 +1,44 @@
+import UserService from '../services/UserService';
+import { isPasswordTrue } from '../utils/paaswordDecrypt';
+import generateToken from '../utils/tokenHandler';
+
+/**
+   * function findAll() returns all users in db
+   * @param {object} req it request
+   * @param {object} res it response
+   * @returns {object} send reponse to the user
+   */
+const getUser = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+
+    const gottenUser = await UserService.findUserByEmail(email);
+
+    if (gottenUser) {
+      const { dataValues } = gottenUser;
+      if (await isPasswordTrue(password, dataValues.password)) {
+        const userInfo = {
+          names: dataValues.names,
+          email: dataValues.email
+        };
+        const token = await generateToken(userInfo);
+        res.status(200).json({
+          message: 'successfully logged in',
+          token
+        });
+      } else {
+        res.status(401).json({
```

Comment:
Could you create a middleware to check both email and password you can call it `credentialChecker` and call it in the route like this `router.post('/api/user/', crendentialChecker, login.userLogin)`, and move all the 401 from controller to that middleware.

Reply:
I refactored my codes to make the function easy but didn't create the middleware, you can check the way I did, and you tell me how you think about it.

Reply:
done